core/src/main/scala/org/apache/spark/sql/CachedDataFrame.scala

```diff
@@ -146,7 +149,7 @@ class CachedDataFrame(session: SparkSession, queryExecution: QueryExecution,
       sessionState.conf.activeSchedulerPool
 
     // Check if it is pruned query, execute it automatically on the low latency pool
-    if ((cachedRDD ne null) && cachedRDD.getNumPartitions <= 2 /* some small number */ &&
+    if ((cachedRDD ne null) && isLowLatencyQuery &&
```

Comment:
Shouldn't the (cachedRDD ne null) check also go into isLowLatencyQuery flag above? I think there are some cases where cachedRDD can be null like insert/update/delete in which case isLowLatencyQuery should be false like here.